numpy/core/src/multiarray/datetime.c

```diff
@@ -2845,6 +2845,16 @@ convert_pyobject_to_timedelta(PyArray_DatetimeMetaData *meta, PyObject *obj,
         *out = NPY_DATETIME_NAT;
         return 0;
     }
+    else if (PyArray_IsScalar(obj, Integer)) {
```

Comment:
Move this branch up, next to the branch handling "raw integers" on line 2661?

Reply:
This causes test failures in my hands -- I'm guessing other types of inputs get drawn in to the branch too early if moved up there.